### PR TITLE
Api change

### PR DESCRIPTION
**Pull request by kkamalov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,8 +16,13 @@ let logoTemplate = require("./templates/logo.handlebars");
 require("script!./node_modules/accessibility-developer-tools/dist/js/axs_testing.js");
 
 class Toolbar {
-    constructor() {
+    constructor(inputData) {
         this.activePlugin = null;
+	this.inputData = inputData;
+	this.plugins = [];
+        plugins.default.map((plugin) => {
```
Comment (marcua) [inline index.js]: spacing?

File: index.js [hunk 2]
```diff
@@ -16,8 +16,13 @@ let logoTemplate = require("./templates/logo.handlebars");
 require("script!./node_modules/accessibility-developer-tools/dist/js/axs_testing.js");
 
 class Toolbar {
-    constructor() {
+    constructor(inputData) {
         this.activePlugin = null;
+	this.inputData = inputData;
```
Comment (marcua) [inline index.js]: spacing?

File: index.js [hunk 3]
```diff
@@ -48,35 +53,17 @@ class Toolbar {
...
+	    });
+
 
         let $plugins = (
-            <div className="tota11y-plugins">
-                {$defaultPlugins}
-                {$experimentalPlugins}
-            </div>
-        );
+			<div className="tota11y-plugins">
```
Comment (marcua) [inline index.js]: spacing

File: index.js [hunk 4]
```diff
@@ -16,8 +16,13 @@ let logoTemplate = require("./templates/logo.handlebars");
 require("script!./node_modules/accessibility-developer-tools/dist/js/axs_testing.js");
 
 class Toolbar {
-    constructor() {
+    constructor(inputData) {
```
Comment (marcua) [inline index.js]: `pluginParameters` ?

File: plugins/base.js [hunk 5]
```diff
@@ -14,9 +14,12 @@ let InfoPanel = require("./shared/info-panel");
 let template = require("../templates/plugin.handlebars");
 
 class Plugin {
-    constructor() {
+    constructor(inputData) {
```
Comment (marcua) [inline plugins/base.js]: `optionalParameters` ?

File: index.js (hunk 4 above)
Comment (marcua) [inline index.js]: a comment like
`pluginParameters Allows us to pass various parameters from a user interface (e.g., what screen size, or which SEO terms to optimize for) to each plugin.

File: plugins/base.js [hunk 6]
```diff
@@ -14,9 +14,12 @@ let InfoPanel = require("./shared/info-panel");
...
         this.panel = new InfoPanel(this.getTitle());
         this.$checkbox = null;
+	this.input = null;
+	this.inputData = inputData;
+	this.errors = this.analyze();
```
Comment (marcua) [inline plugins/base.js]: should we do it in the constructor, or have a loop that explicitly calls it later?

not sure i know which is better

File: plugins/base.js [hunk 7]
```diff
@@ -27,6 +30,10 @@ class Plugin {
         return "";
     }
 
+    addInput(input) {
```
Comment (marcua) [inline plugins/base.js]: what's this?

File: plugins/headings/index.js [hunk 8]
```diff
@@ -70,6 +70,7 @@ const ERRORS = {
 
 class HeadingsPlugin extends Plugin {
     getTitle() {
+	this.analyze();
```
Comment (marcua) [inline plugins/headings/index.js]: spacing

File: plugins/headings/index.js [hunk 9]
```diff
@@ -70,6 +70,7 @@ const ERRORS = {
 
 class HeadingsPlugin extends Plugin {
     getTitle() {
```
Comment (marcua) [inline plugins/headings/index.js]: also why do you have to call analyze again?

File: plugins/title/index.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+let $ = require("jquery");
+let Plugin = require("../base");
+
+let outlineItemTemplate = require("./outline-item.handlebars");
+require("./style.less");
+
+const ERRORS = {
+    MULTIPLE_TITLES(titles) {
+        let _tag = (level) => `<code>${level}</code>`;
```
Comment (marcua) [inline plugins/title/index.js]: spacing

Comment (marcua) [inline plugins/title/index.js]: it looked like you didn't have to quote html/xml, am i right?

File: plugins/title/index.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+	let description = `
```
Comment (marcua) [inline plugins/title/index.js]: same story?

File: plugins/headings/index.js [hunk 12]
```diff
@@ -77,87 +78,58 @@ class HeadingsPlugin extends Plugin {
         return "Highlights headings (<h1>, <h2>, etc) and order violations";
     }
 
-    /**
-     * Computes an outline of the page and reports any violations.
-     */
-    outline($headings) {
-        let $items = [];
+    run() {
```
Comment (marcua) [inline plugins/headings/index.js]: what was the reasoning for moving stuff out of run and into analyze?

File: plugins/title/index.js (hunk 10 above)
Comment (marcua) [inline plugins/title/index.js]: how do you decide _ or not?

Comment (marcua) [inline plugins/title/index.js]: level?

File: plugins/title/index.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+	this.errors.map((error) => {
+		// Register an error to the info panel
+		this.error(error.title,
+			   $(error.description),
+			   $(error.el));
+	    });
+    }
+
+    analyze() {
+        let $titles = $("title");
```
Comment (marcua) [inline plugins/title/index.js]: spacing

File: plugins/title/index.js [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+    getTitle() {
+        return "Title";
+    }
+
+    getDescription() {
+        return "Title Violations";
+    }
+
+    run() {
+	this.errors.map((error) => {
```
Comment (marcua) [inline plugins/title/index.js]: do you have to return the mapping?

File: plugins/title/index.js [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+
+    getTitle() {
+        return "Title";
+    }
+
+    getDescription() {
+        return "Title Violations";
+    }
+
+    run() {
```
Comment (marcua) [inline plugins/title/index.js]: this feels like every Plugin will have the same implementation.  if that's true, it should go in the parent class

File: plugins/headings/index.js [hunk 16]
```diff
@@ -77,87 +78,58 @@ class HeadingsPlugin extends Plugin {
...
+	this.errors.map((error) => {
+		// Register an error to the info panel
+		let infoPanelError = this.error(
+						error.title,
+						$(error.description),
+						error.el);
+		// Place an error label on the heading tag
```
Comment (marcua) [inline plugins/headings/index.js]: do we sitll do this?

File: plugins/headings/index.js (hunk 9 above)
Comment (kkamalov) [inline plugins/headings/index.js]: just forgot to remove.

File: index.js (hunk 4 above)
Comment (kkamalov) [inline index.js]: great :)

File: index.js (hunk 3 above)
Comment (kkamalov) [inline index.js]: fixing all the sapcing. fucked up with my emacs setup

File: plugins/base.js (hunk 5 above)
Comment (kkamalov) [inline plugins/base.js]: sounds good

File: plugins/base.js (hunk 6 above)
Comment (kkamalov) [inline plugins/base.js]: i can go either way. which one do you think it should be?

File: plugins/base.js (hunk 7 above)
Comment (kkamalov) [inline plugins/base.js]: gone. had it from before.

File: plugins/headings/index.js (hunk 12 above)
Comment (kkamalov) [inline plugins/headings/index.js]: run() runs everytime we want to show particular plugin is clicked. we can talk about it in our 1-1

File: plugins/headings/index.js (hunk 16 above)
Comment (kkamalov) [inline plugins/headings/index.js]: it should be here

File: plugins/title/index.js (hunk 10 above)
Comment (kkamalov) [inline plugins/title/index.js]: fixed spacing. 
that's the convention they had they've been using. helper functions within a funciton seem to start with _

File: plugins/title/index.js (hunk 15 above)
Comment (kkamalov) [inline plugins/title/index.js]: i wasnt sure if that statement is true. we can discuss :)

File: plugins/title/index.js (hunk 14 above)
Comment (kkamalov) [inline plugins/title/index.js]: nope.
this.error() renders handleparjs

**Comment by marcua** [comment]
why do we do raw html sometimes and quoted html other times?

**Comment by marcua** [comment]
rename run to prepareForRender

**Comment by marcua** [comment]
consider renaming `this.error`

**Comment by marcua** [comment]
SHIP IT :ship: :shipit:

**Pull request merged**